src/ballet/ed25519/fd_x25519.c

```diff
@@ -155,7 +158,7 @@ fd_x25519_montgomery_ladder( fd_f25519_t *       x2,
     FD_R43X6_QUAD_LANE_SUB_FAST( P, P, 0,1,0,1, P, Q ); // D = x_3 - z_3,            P  = A  |B  |C    |D,     in u45|s44|u45|s44
     FD_R43X6_QUAD_PERMUTE      ( Q, 0,1,1,0, P );       // BB = B^2,                 P  = A  |B  |B    |A,     in u44|u44|u44|u44
     FD_R43X6_QUAD_MUL_FAST     ( P, P, Q );             // DA = D * A,               P  = AA |BB |CB   |DA,    in u62|u62|u62|u62
-    FD_R43X6_QUAD_FOLD_SIGNED  ( P, P );                // DA = D * A,               P  = AA |BB |CB   |DA,    in u44|u44|u44|u44
+    FD_R43X6_QUAD_FOLD_UNSIGNED( P, P );                // DA = D * A,               P  = AA |BB |CB   |DA,    in u44|u44|u44|u44
```

Comment:
For the longest time I couldn't understand why I needed fold_signed after mul. That didn't make any sense, but we were failing one of the wycheproofs tests. Well, the issue was reducing the input. Once the input is reduced, the loop works fine. Bonus: we get a slight perf improvement :)